src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+     *
+     * @return The {@link MongoClientSettings}.
+     * @see Builder#applyToMongoClientSettings(Consumer)
+     */
+    public MongoClientSettings getMongoClientSettings() {
+        return mongoClientSettings;
+    }
+
+    /**
+     * Gets the name of a MongoDB database used as the {@linkplain Connection#getSchema() JDBC schema} of a
+     * {@linkplain Connection connection} {@linkplain MongoConnectionProvider#getConnection() obtained} from
+     * {@link MongoConnectionProvider}.
+     *
+     * @return The name of the default MongoDB database.
+     * @see Builder#databaseName(String)
+     * @see DatabaseMetaData#getSchemaTerm()
+     */
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    /**
+     * The {@linkplain Connection#getAutoCommit() auto-commit mode} of {@linkplain Connection connections}
+     * {@linkplain MongoConnectionProvider#getConnection() obtained} from {@link MongoConnectionProvider}.
+     *
+     * @return The default auto-commit mode.
+     * @see Builder#autoCommit(boolean)
+     */
+    public boolean isAutoCommit() {
+        return autoCommit;
+    }
+
+    /**
+     * Creates a new {@link MongoDialectSettings.Builder} based on {@code configProperties}.
+     *
+     * @param configProperties The {@linkplain Configurable#configure(Map) configuration properties}.
+     * @return A new {@link MongoDialectSettings.Builder}.
+     */
+    public static MongoDialectSettings.Builder builder(Map<String, Object> configProperties) {
+        return new MongoDialectSettings.Builder(notNull("configProperties", configProperties));
+    }
+
+    /** A builder for {@code MongoDialectSettings}. */
+    public static final class Builder {
+        private final MongoClientSettings.Builder mongoClientSettingsBuilder;
+        private @Nullable String databaseName;
+        private boolean autoCommit;
+
+        private Builder(Map<String, Object> configProperties) {
+            mongoClientSettingsBuilder = MongoClientSettings.builder();
+            var connectionString = ConfigPropertiesParser.getConnectionString(configProperties);
+            if (connectionString != null) {
+                mongoClientSettingsBuilder.applyConnectionString(connectionString);
+                databaseName = connectionString.getDatabase();
+            }
+            var autoCommit = ConfigPropertiesParser.getAutoCommit(configProperties);
+            if (autoCommit != null) {
+                this.autoCommit = autoCommit;
+            }
+        }
+
+        /**
+         * Configures {@link MongoDialectSettings}.
+         *
+         * <p>Note that if you {@link MongoClientSettings.Builder#applyConnectionString(ConnectionString)} with
+         * {@linkplain ConnectionString#getDatabase() database name} configured, you still must configure the database
+         * name via {@link Builder#databaseName(String)}, as there is no way for that to happen automatically.
```

Comment:
I like your design and I think it is better than previous implementation.
If I were the common user I would expect the database name in connection string should be the default one if no further db name is configured via current design to override it. but I agree it is not easy to achieve that without keeping the connection string as a field.

Reply:
but if we made the following code changes, it seems we can achieve the goal to avoid double db name specification:
```
private String defaultDatabaseName;

public MongoDialectSettings.Builder applyConnectionString(String connectionString) {
      mongoClientSettingsBuilder.applyConnectionString(connectionString);
      defaultDatabaseName = new ConnectionString(connectionString).getDatabase();
}

// ... use defaultDatabaseName if needed

```